### PR TITLE
docs: add contribution handbook with governance and review guidelines

### PR DESCRIPTION
**Pull request by fruch**
## Description

Add a comprehensive contribution handbook (`docs/contribution-handbook.md`) that documents governance, roles, decision-making processes, code review expectations, and operational guidelines for SCT contributors and maintainers.

The handbook covers:

- **Governance and Roles** — Contributor, Maintainer, and Project lead role definitions, how to become a maintainer, decision-making tiers (low/medium/high impact), CODEOWNERS structure, succession planning, and conflict resolution (including the red-flag/coffee-break mechanism)
- **Code Review and Merging** — Review expectations (correctness, tests, style, security, performance), merge criteria, handling large PRs, backport process, and Conventional Commits conventions
- **Supported Branches and Backports** — Branch naming, backport labels, the automated backport workflow, and what gets backported
- **CI/CD and Test Infrastructure** — What runs on every PR, GitHub Actions, AI agents in CI, label reference, and Jenkins pipeline structure
- **Issue and PR Triage** — Placeholder section for triage process, labels, and stale issue policy (to be detailed)
- **Community and Communication** — Placeholder section for channels, meetings, and onboarding (to be detailed)
- **Security** — Key and secrets management, security-sensitive change review, dependency management, and GitHub Actions security
- **Documentation** — Documentation standards, configuration docs, and AI agent configuration

The handbook is inspired by maintainer guides from the Linux kernel, Kubernetes, CPython, CNCF projects, and GitHub's Open Source Guides. It provides a foundation for transparent, merit-based governance and establishes clear expectations for contributors and maintainers.

Several sections include TODO comments for future expansion (nomination process details, label taxonomy, review turnaround SLA, active release branches, and Code of Conduct). Infrastructure and operations topics will live in a separate operational runbook.

## Testing

N/A — Documentation only, no code changes to test.

## PR pre-checks

- [x] No backport labels needed (documentation)
- [x] No commented-out code

https://claude.ai/code/session_01WZCezGsHuQvFPs2F3Dn3wK

### CODE REVIEW

File: docs/maintainers-handbook.md [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+## 6. Community and Communication
+
+| Topic | Description | Inspiration |
+|-------|-------------|-------------|
+| Communication channels | Where discussions happen (Slack, mailing lists, GitHub Discussions) | Kubernetes community channels |
+| Meeting cadence and format | Regular syncs, agendas, notes, recordings | Kubernetes SIG meetings |
+| Code of conduct | Expected behavior, enforcement, reporting mechanisms | CNCF Code of Conduct |
+| Onboarding new contributors | First steps, mentoring, pairing, documentation pointers | CNCF mentorship programs |
+| Public roadmap | How the project roadmap is communicated and updated | CNCF project planning |
+| External communications | Blog posts, conference talks, social media guidelines | CNCF marketing support for maintainers |
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md:475]: I've just ask cluade to pull in ideas from or OSS codebases, for such a doc, hence somog those sections, public communication isn't a must, even that we do blog post from time to time, and I think it's something we should do more, but it's directly related to the mainterers work

File: docs/maintainers-handbook.md [hunk 2]
```diff
@@ -0,0 +1,571 @@
+# Maintainers Handbook
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Agreed — renamed to `docs/contribution-handbook.md` and retitled "Contribution Handbook"; the intro now states the audience is everyone who contributes, not only maintainers (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+should be re-evaluated (see Succession Planning below).
+
+### 1.2 Contributor Ladder
+
+The contributor ladder defines the path from first-time contributor to maintainer.
+Each level has clear entry requirements so advancement is transparent and merit-based.
+
+| Level | Requirements | Privileges |
+|-------|-------------|------------|
+| **Contributor** | Submit at least 1 merged PR | Can open issues and PRs, review code, participate in discussions |
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Reworked section 1.2 into "How to Become a Maintainer" — dropped the ladder table and the contributor level; it now states everyone who contributes is a contributor with no special status, and only describes how maintainership is earned (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+3. Upon approval, access is granted and the candidate is added to CODEOWNERS and team lists
+
+<!-- TODO: document the nomination process in more detail (where it happens, template, examples) -->
+
+### 1.3 Decision-Making Process
+
+Technical decisions follow a tiered approach based on impact:
+
+**Low impact** (bug fixes, small refactors, test additions) — a single reviewer
+approval is sufficient. The PR author or a maintainer can merge.
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Reworded: once approved, the author or a maintainer can merge, and authors do not merge their own PRs before someone else has reviewed them, except in emergencies (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+### 1.4 OWNERS / CODEOWNERS Files
+
+Code ownership determines who is automatically requested for reviews and who has
+authority over specific areas of the codebase. The current ownership map is at
+[`.github/CODEOWNERS`](../.github/CODEOWNERS).
+
+**How ownership is assigned:**
+- CODEOWNERS entries map file patterns to GitHub teams or individuals
+- Ownership follows expertise — the people who wrote and maintain the code own it
+- Shared ownership (2-3 people per area) prevents single points of failure
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Removed the bullet (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 6]
```diff
@@ -0,0 +1,571 @@
...
+- Owners are automatically added as reviewers on PRs touching their files
+- At least one owner must approve before the PR can be merged
+- Owners are responsible for triaging issues in their area
+- Owners decide the technical direction for their area, within the project's overall architecture
+
+**Example areas and their scope:**
+- `sdcm/cluster_aws.py`, `sdcm/provision/aws/` — AWS backend
+- `sdcm/nemesis.py`, `sdcm/nemesis_registry.py` — Nemesis framework
+- `sdcm/sct_config.py`, `defaults/` — Configuration system
+- `jenkins-pipelines/` — CI/CD pipelines — see [`docs/sct-pipelines.md`](sct-pipelines.md)
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Replaced the `jenkins-pipelines/` entry with `Jenkinsfile` — Jenkins CI for PRs, as suggested (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 7]
```diff
@@ -0,0 +1,571 @@
...
+- `sdcm/cluster_k8s/` — Kubernetes backends — see [`docs/kubernetes_backend.md`](kubernetes_backend.md)
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Removed the Kubernetes backends line from the examples (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 8]
```diff
@@ -0,0 +1,571 @@
...
+
+### 1.5 Succession Planning
+
+Maintainers step down, change roles, or become inactive. The project must handle
+these transitions gracefully to avoid stalled reviews, abandoned areas, and bus-factor
+risks.
+
+**Emeritus status:**
+- Merge access and CODEOWNERS entries are removed
+- The person is acknowledged in a contributors/emeritus list
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: We do not have one, and at this team size we probably do not need it — removed the emeritus-list mention and its TODO; the section now only says merge access and CODEOWNERS entries are removed (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 9]
```diff
@@ -0,0 +1,571 @@
...
+- Emeritus maintainers can return to active status by resuming contributions and going through an expedited nomination
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Removed (6ac6d6a331).

File: docs/contribution-handbook.md [hunk 10]
```diff
@@ -0,0 +1,571 @@
...
+- Acknowledge valid points from the other side
+- Propose compromise solutions when possible
+
+**Level 2 — Involve a third maintainer.** If the two parties cannot reach agreement
+after several rounds of discussion, a third maintainer from the same area (or an
+adjacent area) is asked to weigh in. The third maintainer reviews the arguments and
+either sides with one position or proposes a synthesis.
+
+**Level 3 — Project lead decision.** If Level 2 does not resolve the disagreement,
+the project lead makes a binding decision. The decision is documented on the PR or
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:175]: Added a project lead definition to section 1.1 (makes the final call when consensus cannot be reached, drives succession planning), so the references in 1.3 and 1.6 now point to a defined role (6ac6d6a331).

File: docs/contribution-handbook.md [hunk 11]
```diff
@@ -0,0 +1,571 @@
...
+
+**Bus-factor mitigation:**
+- Every area should have at least 2 owners in CODEOWNERS
+- Critical areas (CI/CD, configuration, core cluster code) should have 3+ owners
+- Regularly review CODEOWNERS for single-owner areas and prioritize finding co-owners
+
+### 1.6 Conflict Resolution
+
+Disagreements are normal and healthy. The project uses a structured escalation path
+to resolve them without damaging relationships.
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:153]: Added a "Red flag / coffee break" paragraph at the top of the conflict-resolution section — any participant can raise a red flag to pause a heated or circular discussion, which then resumes later in a synchronous conversation (6ac6d6a331).

File: docs/contribution-handbook.md [hunk 12]
```diff
@@ -0,0 +1,571 @@
...
+- Process feedback is welcome after the fact ("we should handle this differently next time") but re-litigating decided issues is not
+
+## 2. Code Review and Merging
+
+### 2.1 Code Review Expectations
+
+Every pull request must be reviewed before merging. Reviewers should check for:
+
+- **Correctness** — does the code do what the PR description says? Are edge cases handled?
+- **Test coverage** — are there unit tests for new logic? Integration tests where needed?
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:192]: Added to the test-coverage bullet: for changes automated CI cannot cover (Jenkins pipeline or provisioning changes), manually triggering the affected pipelines may be necessary before merging (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 13]
```diff
@@ -0,0 +1,571 @@
...
+### 2.1 Code Review Expectations
+
+Every pull request must be reviewed before merging. Reviewers should check for:
+
+- **Correctness** — does the code do what the PR description says? Are edge cases handled?
+- **Test coverage** — are there unit tests for new logic? Integration tests where needed?
+  See the [`writing-unit-tests`](../skills/writing-unit-tests/SKILL.md) and
+  [`writing-integration-tests`](../skills/writing-integration-tests/SKILL.md) skills
+  for testing guidance
+- **Style and conventions** — imports at the top (no inline imports), pytest style (not
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Done — the bullet now only references AGENTS.md, without naming any specific conventions in this document (6ac6d6a331).

File: docs/contribution-handbook.md [hunk 14]
```diff
@@ -0,0 +1,571 @@
...
+
+### 2.2 Review Turnaround
+
+Stale PRs slow everyone down. To keep the review pipeline healthy:
+
+- Prioritize reviewing others' PRs over opening new ones
+- If you cannot review a PR assigned to you, reassign it or let the author know
+- Authors should keep PRs small and focused to make reviews easier and faster
+
+<!-- TODO: agree on expected review turnaround (e.g., first response within N business days) -->
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:219]: Added a bullet with the `is:open is:pr review-requested:@me` search filter (linked, scoped to this repo) — it lists PRs awaiting your review and clears once you submit one (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 15]
```diff
@@ -0,0 +1,571 @@
...
+- Authors should keep PRs small and focused to make reviews easier and faster
+
+<!-- TODO: agree on expected review turnaround (e.g., first response within N business days) -->
+
+### 2.3 Merge Criteria
+
+A PR is ready to merge when all of the following are true:
+
+- At least one approval from a reviewer (two for medium/high impact — see section 1.3)
+- CI checks pass (pre-commit, unit tests, any triggered integration tests)
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Yes — documented it: a PR with a failing check may still be merged if the failure was analyzed and is clearly unrelated to the change (e.g., a known flaky test or something already broken on master) (6ac6d6a331).

File: docs/contribution-handbook.md [hunk 16]
```diff
@@ -0,0 +1,571 @@
...
+configuration.
+
+**Format:**
+```
+type(scope): subject
+
+Body explaining what changed and why (minimum 30 characters).
+```
+
+**Allowed types:** `ci`, `docs`, `feature`, `fix`, `improvement`, `perf`, `refactor`,
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:283]: Actually `feat` is not allowed — the type-enum in `commitlint.config.js` (line 11, same on master) only lists `feature`, so a `feat(...)` commit would be rejected by the commitlint hook. Left the list as is; if we want to accept `feat` too, that would be a commitlint config change in a separate PR.

File: docs/maintainers-handbook.md [hunk 17]
```diff
@@ -0,0 +1,571 @@
...
+## 3. Supported Branches and Backports
+
+### 3.1 Branch Naming and Purpose
+
+SCT maintains several branch types, each tied to a ScyllaDB release line:
+
+| Branch pattern | Purpose | Example |
+|---------------|---------|---------|
+| `master` | Main development branch, all new work lands here first | `master` |
+| `branch-X.Y` | Tracks a ScyllaDB OSS release line | `branch-2024.2`, `branch-2025.1` |
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Dropped "OSS" — now just "Tracks a ScyllaDB release line" (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 18]
```diff
@@ -0,0 +1,571 @@
...
+provides the full step-by-step workflow. AI agents (Claude, Copilot) can also
+resolve backport conflicts when tagged on the PR — see section 4.3.
+
+### 3.5 What Gets Backported
+
+- **Bug fixes** — always backport to affected branches
+- **Test stability improvements** — backport when the flaky test affects the branch
+- **New features** — generally do not backport unless the feature is needed for
+  testing a specific release
+- **Refactoring** — do not backport unless it is a prerequisite for a bug fix
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Added: "...or skipping it would make future backports to that branch significantly harder" (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 19]
```diff
@@ -0,0 +1,571 @@
...
+<!-- TODO: document any additional labels used for issue triage or priority -->
+
+### 4.5 Jenkins Pipeline Structure
+
+Jenkins pipelines live in `jenkins-pipelines/` organized by test category.
+See [`docs/sct-pipelines.md`](sct-pipelines.md) for the full overview.
+
+| Directory | Purpose |
+|-----------|---------|
+| `jenkins-pipelines/oss/` | ScyllaDB OSS tests (longevity, upgrade, artifacts, nemesis, etc.) |
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Changed to "General ScyllaDB tests (longevity, upgrade, artifacts, nemesis, etc.) — the `oss` name is historical" (6ac6d6a331).

File: docs/maintainers-handbook.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Removed the "External communications" row — it came from the CNCF maintainer guides and is not relevant for us (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 20]
```diff
@@ -0,0 +1,571 @@
...
+The Claude CI workflows restrict access to org members and collaborators, but
+reviewers should still verify that AI-generated code does not introduce credential
+leaks or injection vulnerabilities.
+
+### 7.3 Dependency Management
+
+Dependencies are managed via `pyproject.toml` and `uv.lock`. Security considerations:
+
+- Review dependency updates for supply chain risks (new or changed transitive dependencies)
+- Pin versions in `uv.lock` to ensure reproducible builds
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Fixed — "Pin versions in `pyproject.toml`; the resolved versions captured in `uv.lock` ensure reproducible builds" (6ac6d6a331).

File: docs/maintainers-handbook.md [hunk 21]
```diff
@@ -0,0 +1,571 @@
...
+| Keeping docs up to date | Process for updating docs when code changes | <!-- TODO: document the doc-update policy --> |
+| Configuration documentation | How config parameters are documented, auto-generation | [`docs/sct-configuration.md`](sct-configuration.md) |
+| Architecture decision records | How and when to document significant technical decisions | [`docs/plans/`](plans/) for implementation plans |
+| Runbooks and playbooks | Operational guides for common maintenance tasks | <!-- TODO: create runbooks directory --> |
+| AI agent configuration | Maintaining `CLAUDE.md`, `AGENTS.md`, and skills that guide AI behavior | [`skills/designing-skills/SKILL.md`](../skills/designing-skills/SKILL.md) |
+
+## 9. Infrastructure and Operations (provisional)
+
+> **Note:** This section may belong in a separate operational runbook rather than
+> the maintainers handbook. Keeping it here for now until we decide on the right home.
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/maintainers-handbook.md]: Removed section 9 entirely — infrastructure/operations topics will live in a separate operational runbook (6ac6d6a331).

**Comment by fruch** [comment]
@claude review

File: docs/contribution-handbook.md [hunk 22]
```diff
@@ -0,0 +1,576 @@
+# Contribution Handbook
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:1]: Already handled — the PR title and description were updated to say "contribution handbook" and reference `docs/contribution-handbook.md`. The rename was pehala's request (the audience is wider than maintainers), so the file name is the intended deliverable and the metadata now matches it.

File: docs/contribution-handbook.md [hunk 23]
```diff
@@ -0,0 +1,576 @@
...
+- Owners are automatically added as reviewers on PRs touching their files
+- At least one owner must approve before the PR can be merged
+- Owners are responsible for triaging issues in their area
+- Owners decide the technical direction for their area, within the project's overall architecture
+
+**Example areas and their scope:**
+- `sdcm/cluster_aws.py`, `sdcm/provision/aws/` — AWS backend
+- `sdcm/nemesis.py`, `sdcm/nemesis_registry.py` — Nemesis framework
+- `sdcm/sct_config.py`, `defaults/` — Configuration system
+- [`Jenkinsfile`](../Jenkinsfile) — Jenkins CI for PRs (see section 4.1)
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Fixed in 557fa89a1d. The list now uses patterns that actually exist in `.github/CODEOWNERS` (`/sdcm/cluster_aws.py`, `/sdcm/provision/aws`, `/defaults`, `/sdcm/nemesis/registry`), is labeled "Example entries" pointing at CODEOWNERS as authoritative, and a following paragraph states explicitly that not every area has an entry — with Jenkins CI for PRs given as an example of a responsibility held by convention rather than by file pattern. (Keeping the Jenkins example was pehala's request; the framing is what changed.)

File: docs/contribution-handbook.md [hunk 24]
```diff
@@ -0,0 +1,576 @@
...
+<!-- TODO: set up automated vulnerability scanning (e.g., Dependabot, Snyk, or Renovate) -->
+
+### 7.4 GitHub Actions Security
+
+Several workflows use elevated permissions or secrets. Key security patterns:
+
+- **Org membership checks** — the Claude workflow verifies the commenter is an org
+  member before executing. The Docker image build workflow checks team affiliation
+- **`pull_request_target` caution** — workflows using this event have access to secrets
+  even for fork PRs. Each such workflow includes a membership gate
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Fixed in 557fa89a1d — good catch, this was a false security claim. The bullet now says only `build-docker-image` gates on membership (it is the one that checks out the PR head), and explains why the others are safe for a different reason: `auto_assign` and `call_jira_sync` only call APIs and never execute fork code, and `test-hydra-macos` checks out the base branch (the `pull_request_target` default) behind a label trigger that requires write access. It closes with the rule that checking out fork code with secrets in scope is the dangerous combination.

File: docs/contribution-handbook.md [hunk 25]
```diff
@@ -0,0 +1,576 @@
...
+
+### 2.4 Handling Large PRs
+
+Large PRs are harder to review and more likely to introduce bugs. When a PR is too
+large:
+
+- Ask the author to split it into smaller, self-contained PRs
+- Each PR should be independently reviewable and testable
+- A good split follows logical boundaries: refactoring in one PR, new feature in
+  another, tests in a third if they are substantial
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Agreed, fixed in 557fa89a1d. The bullet now reads: preparatory refactoring in one PR, the new feature in another, and "keep a behavior change and the tests that cover it in the same PR — only broader test expansion beyond the change itself is worth splitting out". That removes the tension with the independently-testable requirement and the coverage merge criterion.

File: docs/contribution-handbook.md [hunk 26]
```diff
@@ -0,0 +1,576 @@
...
+| `manager-X.Y` | Scylla Manager release line | `manager-3.4` |
+
+Branch creation follows ScyllaDB release cycles — when a new ScyllaDB version
+is branched, a corresponding SCT branch is created to track it.
+
+### 3.2 Backport Labels
+
+Every PR targeting `master` **must** carry a backport label. This is enforced by
+the [`pr-require-backport-label`](../.github/workflows/pr-require-backport-label.yaml)
+GitHub Action. The required label must match one of:
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Verified and fixed in 557fa89a1d. The job has `if: github.event.pull_request.draft == false` and subscribes only to `[opened, labeled, unlabeled, synchronize]`, so a PR opened as a draft and later marked ready without any further push or label change is never checked. Rather than assert enforcement, the text now describes the actual behavior and tells contributors to apply the label themselves instead of relying on the check. Adding `ready_for_review`/`reopened` to the workflow would be a separate PR.

File: docs/contribution-handbook.md [hunk 27]
```diff
@@ -0,0 +1,576 @@
...
+
+| Label | Meaning |
+|-------|---------|
+| `backport/none` | This change does not need backporting |
+| `backport/X.Y` | Backport to `branch-X.Y` (e.g., `backport/2025.1`) |
+| `backport/perf-vX` | Backport to `branch-perf-vX` |
+| `backport/manager-X.Y` | Backport to `manager-X.Y` |
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Partly incorrect, so I did not apply this as written — two things to separate:

1. **The label does exist in versioned form.** `gh label list` shows `backport/manager` *and* `backport/manager-3.10` / `backport/manager-3.11` (plus their `-done` counterparts), so documenting only the generic label would be less accurate, not more.
2. **The `branch-manager-X.Y` translation never happens.** `auto-backport.py` filters labels first with `backport_label_pattern = re.compile(r"backport/(perf-*.\d+$|\d+\.\d+$)")` (line 22, applied at line 171). Neither `backport/manager` nor `backport/manager-3.11` matches it, so they are dropped before reaching the `replace("backport/", "branch-")` on line 191 — no PR is created against a nonexistent branch.

What was genuinely wrong in the doc was implying automation covers Manager, so 557fa89a1d adds an "Automated?" column marking the Manager labels **manual**, lists both label forms, and states that the matcher does not match them so Manager backports are cherry-picked and opened by hand.

File: docs/contribution-handbook.md [hunk 28]
```diff
@@ -0,0 +1,603 @@
...
+
+When a PR with a `backport/X.Y` label is merged to `master`, the
+[`add-label-when-promoted`](../.github/workflows/add-label-when-promoted.yaml)
+workflow triggers the automated backport process
+([`.github/scripts/auto-backport.py`](../.github/scripts/auto-backport.py)):
+
+1. The script cherry-picks the PR's commits to the target branch
+2. A backport PR is created automatically (authored by `scylladbbot`)
+3. If cherry-pick conflicts occur, the backport PR is created as a **draft** with
+   a `conflicts` label
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Verified and largely applied in 4cec5a29b5 — thanks, this one was only visible after the branch got rebased onto current master.

Confirmed both workflows are live with overlapping triggers: `add-label-when-promoted.yaml` (push to `master`/`branch-*.*`/`branch-perf-v*`/`manager-*.*` plus `pull_request_target: [labeled]`) and `call_backport_with_jira.yaml` (push to the same set plus `pull_request_target: [labeled, closed]`), and the latter's header comment says outright that it replaces the former plus `auto-backport.py`.

Doc changes: section 3.3 now documents `call_backport_with_jira` and the shared `scylladb/github-automation` workflow as the backport automation (including the Jira sub-task step and the `cascade_backport` opt-in), the workflow table lists it and marks `add-label-when-promoted` as legacy/superseded, and a note records that the legacy workflow still triggers on overlapping events so backports can currently be created twice.

I did **not** remove or disable the legacy workflow: this is a docs-only PR, and deleting live backport automation is a change its owners should make deliberately. The note points at it so it is not silently forgotten.

File: docs/contribution-handbook.md [hunk 29]
```diff
@@ -0,0 +1,603 @@
...
+  the base repository and can access secrets even for fork PRs. Only
+  `build-docker-image` gates on membership, because it is the one that checks out and
+  builds the PR head. The others are safe for a different reason: they never execute
+  fork code — [`auto_assign`](../.github/workflows/auto_assign.yaml) and
+  [`call_jira_sync`](../.github/workflows/call_jira_sync.yml) only call APIs, and
+  [`test-hydra-macos`](../.github/workflows/test-hydra-macos.yaml) checks out the base
+  branch (the `pull_request_target` default) and is triggered by a label, which requires
+  write access to apply. When adding or changing a `pull_request_target` workflow, keep
+  its `permissions` block minimal, and if it must check out the PR head, gate it on
+  membership — checking out fork code with secrets in scope is the dangerous combination
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Good refinement — applied in 4cec5a29b5. You are right that "does not execute fork code" is too coarse a criterion to call a workflow safe, and `call_jira_sync` is exactly the counterexample: it passes PR-supplied text (the Jira key from the PR body) into authenticated Jira mutations.

Section 7.4 now states there is no single control that makes `pull_request_target` safe and assesses each workflow separately: `build-docker-image` (checks out the PR head → gated on `dev` team affiliation), `test-hydra-macos` (AWS secrets, but base-branch checkout behind a write-only label), `call_jira_sync` (no fork code, but untrusted input reaching a credentialed API — bounded by the Jira account's permissions), and `auto_assign` (GitHub API only, no other secrets). The closing rule now also requires validating any PR-supplied value before it reaches an authenticated API.

File: docs/contribution-handbook.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md]: Correction to my previous reply on this thread — I was reasoning against the legacy automation, and the current one behaves differently.

After the branch was rebased onto current master, `.github/workflows/call_backport_with_jira.yaml` is the active backport automation (it replaces `add-label-when-promoted.yaml` + `auto-backport.py`), and it explicitly handles Manager branches: *"Manager branches: manager-X.Y, driven by backport/manager-X.Y labels."* So `backport/manager-X.Y` **is** automated, and my earlier "manual" claim was wrong.

What still stands: the versioned labels do exist (`backport/manager-3.10`, `backport/manager-3.11`), so documenting only a generic `backport/manager` would be less accurate. And the `branch-manager-X.Y` mistranslation you described never occurs — in the legacy script the label is filtered out by `backport_label_pattern` before reaching the `replace("backport/", "branch-")`, and the current shared workflow maps `backport/manager-X.Y` to `manager-X.Y` directly.

4cec5a29b5 replaces the "Automated?" column with a "Target branch" column showing `backport/manager-X.Y` → `manager-X.Y`, and drops the manual-cherry-pick paragraph entirely.

File: docs/contribution-handbook.md [hunk 30]
```diff
@@ -0,0 +1,628 @@
...
+  entries are removed
+
+**Planned transitions:**
+- Knowledge transfer includes: undocumented context, ongoing work, known technical debt
+- A transition period with overlapping access is recommended
+
+**Bus-factor mitigation:**
+- Every area should have at least 2 owners in CODEOWNERS
+- Critical areas (CI/CD, configuration, core cluster code) should have 3+ owners
+- Regularly review CODEOWNERS for single-owner areas and prioritize finding co-owners
```
Comment (pehala) [inline docs/contribution-handbook.md:148]: This is nice to read, but I dont think we adhere to it.

File: docs/contribution-handbook.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (pehala) [reply docs/contribution-handbook.md:283]: https://github.com/scylladb/scylla-cluster-tests/pull/15368/changes/5c195aade3274877b08fd5814ecc1cfc1b7344ef IT did not fail on my feat commits.

[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:283]: show me the PR it's part of it, I don't know what was working install when you were working locally

File: docs/contribution-handbook.md (hunk 30 above)
Comment (fruch) [reply docs/contribution-handbook.md:148]: last time someone left, we didn't had CODEOWNERS file.
it's something we should be doing, hence why writing it down.

File: docs/contribution-handbook.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:283]: I would guess the commitmsg hook isn't installed

File: docs/contribution-handbook.md [hunk 31]
```diff
@@ -0,0 +1,628 @@
...
+<!-- TODO: document the nomination process in more detail (where it happens, template, examples) -->
+
+### 1.3 Decision-Making Process
+
+Technical decisions follow a tiered approach based on impact:
+
+**Low impact** (bug fixes, small refactors, test additions) — a single reviewer
+approval is sufficient. Once approved, any maintainer can merge — including the
+author, if they are a maintainer. Authors do not merge their own PRs before
+someone else has reviewed them, except in emergencies (e.g., CI is broken for
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:76]: that's o.k.

File: docs/contribution-handbook.md [hunk 32]
```diff
@@ -0,0 +1,628 @@
...
+(highest-to-lowest) backports instead.
+
+### 3.3 Automated Backport Workflow
+
+When a PR carrying a `backport/*` label is merged, the
+[`call_backport_with_jira`](../.github/workflows/call_backport_with_jira.yaml)
+workflow creates the backport PRs. It is a thin caller for the shared reusable
+workflow in
+[`scylladb/github-automation`](https://github.com/scylladb/github-automation),
+which handles all three target-branch families above:
```
[the remark this answers is not in the document]
Comment (fruch) [reply docs/contribution-handbook.md:354]: that fine

File: docs/contribution-handbook.md (hunk 30 above)
Comment (pehala) [reply docs/contribution-handbook.md:148]: I meant having multiple owners for a section, I believe it is sometimes only one person

File: docs/contribution-handbook.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (pehala) [reply docs/contribution-handbook.md:283]: https://github.com/scylladb/scylla-cluster-tests/pull/15368
